spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -640,20 +637,31 @@ case class DeltaSharingSource(
       }
     }
 
-    if (!tableFiles.removeFiles.isEmpty) {
+    val addFiles = tableFiles.addFiles
+    if (tableFiles.removeFiles.nonEmpty) {
+      /** A check on the source table that disallows changes on the source data. */
+      val shouldAllowChanges = options.ignoreChanges || skipChangeCommits
```

Comment:
Here are some thoughts behind this:

Since we need to do some adjustment to the variable to reflect this change (adding `|| skipChangeCommits`), the variable names at L131 (`ignoreChanges` and `ignoreDeletes`) doesn't make sense anymore since they are not the same as `option.ignoreChanges` and `option.ignoreDeletes`. Also they don't have to be global variables since they are only used in this method, so converted them as local variables and updated variable names.

We have made a similar change to DeltaSource in this PR: https://github.com/databricks/runtime/pull/56494. I think it's better to keep it consistent here.